contingency-api/src/main/java/eu/itesla_project/contingency/tasks/ContingencyTopologyTraverser.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package eu.itesla_project.contingency.tasks;
+
+import eu.itesla_project.iidm.network.*;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * @author Mathieu Bague <mathieu.bague@rte-france.com>
+ */
+class ContingencyTopologyTraverser {
+
+    private static boolean isOpenable(Switch aSwitch) {
+        return (!aSwitch.isOpen() &&
+                !aSwitch.isFictitious() &&
+                aSwitch.getKind() == SwitchKind.BREAKER);
+    }
+
+    static void traverse(Terminal terminal, Set<Switch> switchesToOpen, Set<Terminal> terminalsToDisconnect) {
+        Objects.requireNonNull(terminal);
+        Objects.requireNonNull(switchesToOpen);
+        Objects.requireNonNull(terminalsToDisconnect);
+
+        terminal.traverse(new VoltageLevel.TopologyTraverser() {
+            @Override
+            public boolean traverse(Terminal terminal, boolean connected) {
+                boolean traverse = true;
+
+                if (terminal.getVoltageLevel().getTopologyKind() == TopologyKind.BUS_BREAKER) {
```

Comment:
This test probably means we have a flaw in traverser design (node/breaker + bus/branch support)